app.py

```diff
@@ -1,17 +1,286 @@
+        composite_cloudfree = ee.Algorithms.Landsat.simpleComposite(**{
+            'collection': collection,
+            'percentile': 75,
+            'cloudScoreRange': 5
+        })
+        return composite_cloudfree.clip(self.roi)
+
+    def _load_indices(self, indices_file):
+        # Read index configurations
+        with open(indices_file, 'r') as stream:
+            try:
+                return yaml.safe_load(stream)
+            except yaml.YAMLError as e:
+                print(e)
+                return None
+        
+    def show_map(self, map=None):
+        if map is not None:
+            self.map = map
+            self.show = True
+
+    def disable_map(self):
+        self.show = False
+    
+    def generate_index(self, index_config):
+        """
+        Generates an index based on the provided index configuration.
+
+        Args:
+            index_config (dict): Configuration for generating the index.
+
+        Returns:
+            ee.Image: The generated index clipped to the region of interest.
+        """
+        match index_config["gee_type"]:
+            case 'image':
+                dataset = ee.Image(index_config['gee_path']).clip(self.roi)
+                if index_config.get('select'):
+                    dataset = dataset.select(index_config['select'])
+            case 'image_collection':
+                dataset = ee.ImageCollection(index_config['gee_path']).filterBounds(self.roi).map(lambda image: image.clip(self.roi)).mean()
+                if index_config.get('select'):
+                    dataset = dataset.select(index_config['select'])
+            case 'feature_collection':
+                dataset = ee.Image().float().paint(ee.FeatureCollection(index_config['gee_path']), index_config['select']).clip(self.roi)
+            case 'algebraic':
+                image = self._cloudfree(index_config['gee_path'])
+                dataset = image.normalizedDifference(['B4', 'B3'])
+            case _:
+                dataset=None
+
+        if not dataset:
+            raise Exception("Failed to generate dataset.")
+        if self.show and index_config.get('show'):
+            map.addLayer(dataset, index_config['viz'], index_config['name'])
+        print(f"Generated index: {index_config['name']}")
+        return dataset
+
+    def zonal_mean_index(self, index_key):
+        index_config = self.indices[index_key]
+        dataset = self.generate_index(index_config)
+        # zm = self._zonal_mean(single, index_config.get('bandname') or 'constant')
+        out = dataset.reduceRegion(**{
+                'reducer': ee.Reducer.mean(),
+                'geometry': self.roi,
+                'scale': 200 # map scale
+            }).getInfo()
+        if index_config.get('bandname'):
+            return out[index_config.get('bandname')]
+        return out
+
+    def generate_composite_index_df(self, indices=[]):
+        data={
```

Comment:
To-do:
1. Introduce `coefficient` field to this json that turns into a dataframe.
2. Introduce in `indices.yaml` coefficients for each component layer of the index.
3. Introduce coefficients in score calculation them . e.g. `Index = C_1 * Layer_1 + C_2 * Layer_2 ...`.